test-data/unit/check-isinstance.test

```diff
@@ -1443,6 +1443,28 @@ else:
 [builtins fixtures/isinstancelist.pyi]
 
 
+[case testIssubclasDestructuringUnions]
+from typing import Union, List, Tuple, Dict, Type
+def f(x: Union[Type[int], Type[str], Type[List]]) -> None:
```

Comment:
@pkch 
>  I thought ``Type`` requires only simple types inside, not constructs like ``Union``

``Union`` is legal as a special case, see for example [typing docs](https://docs.python.org/3/library/typing.html#typing.Type) (emphasis my):
> The only legal parameters for ``Type`` are classes, _unions of classes_, and ``Any``.

> But it 50% works...

Hmm, these two ``Type[Union[int, str]]`` and ``Union[Type[int], Type[str]]`` should be equivalent I think. Is it easy to fix?